### ISSUE TITLE
Why the cache is empty?

### ISSUE THREAD

**Issue by highacid** [engintron/engintron#115]
Hello, 

I have this in status:

Nginx Cache/Temp Disk Usage:
0	/tmp/engintron_dynamic
0	/tmp/engintron_static
0	/tmp/engintron_temp

Any idea why is it empty? Shouldn't it cache something by already running more then 1 week.

**Comment by BomBom123** [comment]
Also experiencing this with over 50 different websites.
Been running for a day or so now.

Nginx Cache/Temp Disk Usage:
4.0K	/tmp/engintron_dynamic
4.0K	/tmp/engintron_static
4.0K	/tmp/engintron_temp

**Comment by fevangelou** [comment]
Are you both using CentOS 7?

**Comment by WebEndev** [comment]
I also have this:

**Nginx Cache/Temp Disk Usage:**
4.0K	/tmp/engintron_dynamic
4.0K	/tmp/engintron_static
4.0K	/tmp/engintron_temp

Been running for about 4 days, with two sites on the same VPS. This seems quite low?

I am running Centos 7.2.1511, on Apache 2.4.18

Thank you.

**Comment by fevangelou** [comment]
Yes, this is indeed weird because proxy caching actually works, and I can verify this only happens with CentOS 7. I have asked the Nginx team to shed some light.

**Comment by BomBom123** [comment]
I am indeed using CentOS 7 and CloudLinux 7.2

**Comment by highacid** [comment]
I am also using CentOS 7.

**Comment by frilogg** [comment]
Im also using CentOS 7... ands this is a problem...

**Comment by fevangelou** [comment]
@frilogg It's not a problem. Nginx caching DOES work. I've already contacted the Nginx team to see if it's a bug with CentOS 7.

**Comment by fevangelou** [comment]
So, thanks to the kind Nginx folks (@LinuxJedi in particularly) it seems this issue is related to systemd being present in CentOS 7 only. The Nginx cache/temp folders exist inside /tmp but essentially in some aliased location. This can easily be disabled, which I already have on my to-do list for v1.7.0 which will also include EA4 support.

Read more about why this occurs on CentOS 7 with systemd here:
https://securityblog.redhat.com/2014/04/09/new-red-hat-enterprise-linux-7-security-feature-privatetmp/
http://draal.com.ua/2014/12/27/nginx-upload-module-php-fpm-systemd-privatetmp-incompatibility/

**Comment by highacid** [comment]
Hello,

```
root@panel [~]# cd /etc/systemd/system/
root@panel [/etc/systemd/system]# ls
./                   cpanellogd.service         cpdavd.service       default.target@        exim.service         ipaliases.service         p0f.service         ror*                     spamd.service                tailwatchd.service
../                  cpanel_php_fpm.service     cpgreylistd.service  default.target.wants/  fastmail.service     mailman.service           pure-authd.service  securetmp.service        stunnel.service
basic.target.wants/  cpanelquotaonboot.service  cphulkd.service      dnsadmin.service       getty.target.wants/  mariadb.service.d/        pure-ftpd.service   smtpmailgidonly.service  sysinit.target.wants/
clamd.service*       cpanel.service             cpipv6.service       dovecot.service        httpd.service        multi-user.target.wants/  queueprocd.service  sockets.target.wants/    system-update.target.wants/
root@panel [/etc/systemd/system]#
```
I am unable to find the path described in the above articles for nginx. Any workaround? When can we expect version 1.7.0

**Comment by fevangelou** [comment]
In CentOS the file is located here: /usr/lib/systemd/system/nginx.service

In there, change PrivateTmp=true to PrivateTmp=false and you're ready.

**Comment by BomBom123** [comment]
Changed.

Also needed these commands running.
systemctl daemon-reload
service nginx restart

**Comment by BomBom123** [comment]
Woo.

Nginx Cache/Temp Disk Usage:
23M	/tmp/engintron_dynamic
42M	/tmp/engintron_static
4.0K	/tmp/engintron_temp

**Comment by WebEndev** [comment]
Hello

This issue is still not resolved. Was it going to be resolved with the release of version 1.70?

**Comment by highacid** [comment]
@WebEndev I hope to be solved in 1.7.0. Waiting for next release, until a manual workaround above described works as designed.

**Comment by fevangelou** [comment]
Yes, it will be solved with v1.7.0.
